terminal.cpp

```diff
@@ -788,6 +788,17 @@ void Terminal::ansiSequence(const QString& seq)
         }
         break;
 
+    case 'Z': // back tab
+        if(cursorPos().y() > 0) {
```

Comment:
```suggestion
        if(cursorPos().y() > 0 && cursorPos().y() <= iTabStops.size()) {
```
It would be good to extend the test to avoid overflow as well. It's not clear whether it's strictly necessary, but this check is also performed elsewhere in similar code, so better safe than sorry.